pkgs/native_assets_cli/lib/src/model/build_output.dart

```diff
@@ -48,12 +48,19 @@ final class BuildOutputImpl implements BuildOutput {
   static const _timestampKey = 'timestamp';
   static const _versionKey = 'version';
 
-  factory BuildOutputImpl.fromYamlString(String yaml) {
-    final yamlObject = loadYaml(yaml);
-    return BuildOutputImpl.fromYaml(as<YamlMap>(yamlObject));
+  factory BuildOutputImpl.fromJsonString(String jsonString) {
+    final Object? json;
+    if (jsonString.startsWith('{')) {
+      json = jsonDecode(jsonString);
+    } else {
+      // TODO(https://github.com/dart-lang/native/issues/1000): At some point
+      // remove the YAML fallback.
+      json = loadYaml(jsonString);
+    }
+    return BuildOutputImpl.fromJson(as<Map<Object?, Object?>>(json));
   }
 
-  factory BuildOutputImpl.fromYaml(YamlMap yamlMap) {
+  factory BuildOutputImpl.fromJson(Map<Object?, Object?> yamlMap) {
```

Comment:
Again variable names - a global search for `yaml` should do the trick to find all instances :)